### PR TITLE
[Instrumentation.SqlClient] Improve changelog

### PR DESCRIPTION
**Pull request by Kielek**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1559#issuecomment-1932666176
Design discussion issue #

## Changes

Better changelog entry

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

**Comment by Kielek** [comment]
@julealgon, please review if it is helpful. Similar changes as you described in EFCore package in contrib repository.

**Comment by Kielek** [comment]
PR for EFCore: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1575

@cijothomas, FYI

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
   `SqlClientTraceInstrumentationOptions`.
   ([#5285](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5285))
 * **Breaking Change**: Stop emitting `db.statement_type` attribute.
+  This attribute never was part of the [sematnic convention](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/database/database-spans.md#call-level-attributes).
```
Comment (Kielek) [inline src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md:15]:
```suggestion
  This attribute never was part of the [semantic convention](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/database/database-spans.md#call-level-attributes).
```

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
   `SqlClientTraceInstrumentationOptions`.
   ([#5285](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5285))
 * **Breaking Change**: Stop emitting `db.statement_type` attribute.
+  This attribute never was part of the [semantic convention](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/database/database-spans.md#call-level-attributes).
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.SqlClient/CHANGELOG.md:15]:
```suggestion
  This attribute never was part of the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/database/database-spans.md#call-level-attributes).
```

**Pull request merged**